### PR TITLE
UE4-Substance compatibility: combined OcclusionRoughnessMetalness map support.

### PR DESCRIPTION
**Pull request by bhouston**
Implement idea discussed and agreed to here:

https://github.com/mrdoob/three.js/issues/10691#issuecomment-276545010

I've updated the webgl_materials_standard example to use a combined Roughness Metalness map as a test of this new efficiency-oriented feature.

### CODE REVIEW

File: src/renderers/shaders/ShaderChunk/aomap_fragment.glsl [hunk 1]
```diff
@@ -1,5 +1,6 @@
 #ifdef USE_AOMAP
 
+	// reads channel R, intended to be used with a combined OcclusionRoughnessMetallic (RGB) texture
```
Comment (WestLangley) [inline src/renderers/shaders/ShaderChunk/aomap_fragment.glsl]: // reads channel R, compatible with an Occlusion/Roughness/Metalness (RGB) texture

I would prefer to say it is "compatible". Saying it is "intended" sounds like a requirement.

**Comment by mrdoob** [comment]
Hmm, `Cerberus_RM.jpg`...

Isn't jpeg compression mixing roughness/metalness? I wonder if it wouldn't be better to keep it as png?

**Comment by mrdoob** [comment]
Also, the title says... OcclusionRoughnessMetalness. Isn't it supposed to be OcclusionMetalnessRoughness? Just double checking...

**Comment by bhouston** [comment]
> Isn't jpeg compression mixing roughness/metalness?
> I wonder if it wouldn't be better to keep it as png?

You are right it will mix them a bit.  I am personally okay with it, if the differences are not perceptible.

Here is the difference with the original roughness map:

image

Here is the difference with the original metalness map:

image

I think the reason for the larger difference with metalness is that JPEG favors green because of is dominance in the luminance equation: https://en.wikipedia.org/wiki/Relative_luminance

**Comment by bhouston** [comment]
BTW some stats:

Original metalness map: 0.721 MB
Original roughness map: 1.47 MB
Combined metalness/roughness map as jpeg: 0.907 MB
Combined metalness/roughness map as png: 7.4 MB

This change saves ~1 MB at a very minor quality difference.  I'd take that any day. :)

**Comment by mrdoob** [comment]
Thanks!

**Comment by WestLangley** [comment]
The docs need to be updated to indicate the channel that is read for each map type.

**Pull request merged**